llama_index/llms/nvidia_tensorrt.py

```diff
@@ -0,0 +1,352 @@
+        if model_path is not None:
+            if not os.path.exists(model_path):
+                raise ValueError(
+                    "Provided model path does not exist. "
+                    "Please check the path or provide a model_url to download."
+                )
+            else:
+                engine_dir = model_path
+                engine_dir_path = Path(engine_dir)
+                config_path = engine_dir_path / "config.json"
+
+                # config function
+                with open(config_path) as f:
+                    config = json.load(f)
+                use_gpt_attention_plugin = config["plugin_config"][
+                    "gpt_attention_plugin"
+                ]
+                remove_input_padding = config["plugin_config"]["remove_input_padding"]
+                tp_size = config["builder_config"]["tensor_parallel"]
+                pp_size = config["builder_config"]["pipeline_parallel"]
+                world_size = tp_size * pp_size
+                assert (
+                    world_size == tensorrt_llm.mpi_world_size()
+                ), f"Engine world size ({world_size}) != Runtime world size ({tensorrt_llm.mpi_world_size()})"
+                num_heads = config["builder_config"]["num_heads"] // tp_size
+                hidden_size = config["builder_config"]["hidden_size"] // tp_size
+                vocab_size = config["builder_config"]["vocab_size"]
+                num_layers = config["builder_config"]["num_layers"]
+                num_kv_heads = config["builder_config"].get("num_kv_heads", num_heads)
+                paged_kv_cache = config["plugin_config"]["paged_kv_cache"]
+                if config["builder_config"].get("multi_query_mode", False):
+                    tensorrt_llm.logger.warning(
+                        "`multi_query_mode` config is deprecated. Please rebuild the engine."
+                    )
+                    num_kv_heads = 1
+                num_kv_heads = (num_kv_heads + tp_size - 1) // tp_size
+
+                self._model_config = ModelConfig(
+                    num_heads=num_heads,
+                    num_kv_heads=num_kv_heads,
+                    hidden_size=hidden_size,
+                    vocab_size=vocab_size,
+                    num_layers=num_layers,
+                    gpt_attention_plugin=use_gpt_attention_plugin,
+                    paged_kv_cache=paged_kv_cache,
+                    remove_input_padding=remove_input_padding,
+                )
+
+                assert (
+                    pp_size == 1
+                ), "Python runtime does not support pipeline parallelism"
+                world_size = tp_size * pp_size
+
+                runtime_rank = tensorrt_llm.mpi_rank()
+                runtime_mapping = tensorrt_llm.Mapping(
+                    world_size, runtime_rank, tp_size=tp_size, pp_size=pp_size
+                )
+                torch.cuda.set_device(runtime_rank % runtime_mapping.gpus_per_node)
```

Comment:
it seems it can only run on GPU not CPU?
should we check the device before running this step.
if there is no gpu, we should raise errors